File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.go
func productExceptSelf(nums []int) []int {
	
	length := len(nums)
	res := make([]int,length)

	res[0] = 1

	i := 1

	for i < length {
		res[i] = res[i - 1] * nums[i - 1]
		i++
	}

	i = i - 2

	
	p := 1

	for i >= 0 {
		p = nums[i + 1] * p
		res[i] = p * res[i]
		i--
	}


	return res
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        
        res = []
        combination = []

        def dfs(i):
            if i >= len(nums) or sum(combination) > target:
                return
            if sum(combination) == target:
                res.append(combination[:])
                return

            dfs(i+1)
            combination.append(nums[i])
            dfs(i)
            combination.pop()

        dfs(0)

        return res